wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CDS\Wpml\Api;
+
+use CDS\Wpml\Post;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Post;
+use WP_Error;
+
+class Endpoints extends BaseEndpoint
+{
+    protected static $instance;
+    protected Post $post;
+
+    public function __construct()
+    {
+        $this->post = Post::getInstance();
+
+        parent::__construct();
+    }
+
+    public static function getInstance(): Endpoints
+    {
+        is_null(self::$instance) and self::$instance = new self();
+
+        return self::$instance;
+    }
+
+    public function hasPermission(): bool
+    {
+        return true; // @TODO: determine permission
```

Comment:
This needs to be updated before we merge

Reply:
Can check if the nonce check happens here too

Reply:
👍